shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -236,3 +238,182 @@ async fn test_hot_reload_with_zero_connections() {
+        .await;
+
+    // Create a new TLS client configured with certificate
+    let client_new =
+        create_tls_valkey_client_from_certs("127.0.0.1", 6380, "/tmp/shotover-cert-test-new");
+    let mut con_new = client_new.get_connection().unwrap();
+
+    // Verify data persistence across certificate change
+    let value: String = con_new.get("cert_key_0").unwrap();
+    assert_eq!(value, "cert_value_0");
+
+    // Verify the new connection works with the new certificate
+    let pong: String = redis::cmd("PING").query(&mut con_new).unwrap();
+    assert_eq!(pong, "PONG");
+
+    // After the new shotover starts, gradual shutdown begins immediately with the first drain.
+    // Subsequent drains happen every 10 seconds.
+    // Sleep 5s to position ourselves in the middle of the time between 1st and 2nd drain.
+    // This reduces race conditions since we're furthest from drain boundaries.
+    tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
+
+    // Verify that old connections with old certificates still work during drain
+    // The old instance should still be serving requests with the old certificate
+    let mut old_connections_working = 0;
+    for con in connections.iter_mut() {
+        if con.get::<_, String>("cert_key_0").is_ok() {
+            old_connections_working += 1;
+        }
+    }
+
+    // At least some old connections should still be working
+    assert!(
+        old_connections_working > 0,
+        "Some old certificate connections should still be working during gradual drain"
+    );
+
+    // Verify new certificate client continues to work throughout the transition
+    // Counter starts at 0, this will increment to 1
+    assert_valkey_connection_works(&mut con_new, Some(1), &[("cert_key_0", "cert_value_0")])
+        .unwrap();
+
+    // Wait for multiple drain cycles to observe gradual connection closure
+    tokio::time::sleep(tokio::time::Duration::from_secs(20)).await;
+
+    // Check that more old connections have been drained
+    let mut old_connections_still_working = 0;
+    for con in connections.iter_mut() {
+        if con.get::<_, String>("cert_key_0").is_ok() {
+            old_connections_still_working += 1;
+        }
+    }
+
+    // Should have fewer working connections after additional drain cycles
+    assert!(
+        old_connections_still_working < old_connections_working,
+        "Old connections should be gradually drained. Before: {}, After: {}",
+        old_connections_working,
+        old_connections_still_working
+    );
+
+    // Verify new certificate client is still working perfectly
+    // Counter is now at 1, this will increment to 2
+    assert_valkey_connection_works(&mut con_new, Some(2), &[("cert_key_0", "cert_value_0")])
+        .unwrap();
+
+    // Create another new client to ensure new connections use the new certificate
+    let client_new2 =
+        create_tls_valkey_client_from_certs("127.0.0.1", 6380, "/tmp/shotover-cert-test-new");
```

Comment:
The new and new2 client seem identical, should we just create a new connection here and reuse `client_new`?